src/v/raft/tests/raft_reconfiguration_test.cc

```diff
@@ -149,6 +148,33 @@ wait_for_offset(model::offset expected, raft_fixture::raft_nodes_t& nodes) {
     co_return result<model::offset>(raft::errc::timeout);
 }
 
+ss::future<result<model::offset>> wait_for_offset(
+  model::offset expected,
+  std::vector<model::node_id> ids,
+  raft_fixture& fixture) {
+    auto start = model::timeout_clock::now();
+    // wait for visible offset to propagate
+    while (start + 10s > model::timeout_clock::now()) {
+        bool aligned = std::all_of(
+          ids.begin(), ids.end(), [&](model::node_id id) {
+              auto& rni = fixture.node(id);
+              vlog(
+                test_log.info,
+                "node: {}, last_visible_index: {}, expected_offset: {}",
+                id,
+                rni.raft()->last_visible_index(),
+                expected);
+              return rni.raft()->last_visible_index() >= expected;
+          });
+
+        if (aligned) {
+            co_return expected;
```

Comment:
Looks a bit weird that on success we are basically returning the input parameter. Why can't we just return the error code?